modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2MultiAttributeUserFilterTestCase.java

```diff
@@ -0,0 +1,396 @@
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.SERVER_URL;
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.TYPE_PARAM;
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.USER_NAME_ATTRIBUTE;
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.VALUE_PARAM;
+
+public class SCIM2MultiAttributeUserFilterTestCase extends ISIntegrationTest {
+
+    private static final List<String> GROUPNAMES = new ArrayList<>(Arrays.asList("scim2Group1", "scim2Group2"));
+
+    private static final List<String> FAMILY_NAME_CLAIM_VALUES = new ArrayList<>(
+            Arrays.asList("scim1user", "scim2user", "scim3user", "scim4user", "scim5user"));
+    private static final List<String> GIVEN_NAME_CLAIM_VALUES = new ArrayList<>(
+            Arrays.asList("user1", "user2", "user3", "user4", "user5"));
+    private static final List<String> EMAIL_TYPE_WORK_CLAIM_VALUES = new ArrayList<>(
+            Arrays.asList("user1@wso2.com", "user2@wso2.com", "user3@wso2.com", "user4@wso2.com", "user5@wso2.com"));
+    private static final List<String> EMAIL_TYPE_HOME_CLAIM_VALUES = new ArrayList<>(
+            Arrays.asList("scim1@gmail.com", "scim2@gmail.com", "scim3@gmail.com", "scim4@gmail.com",
+                    "scim5@gmail.com"));
+    public static final List<String> USERNAMES = new ArrayList<>(
+            Arrays.asList("scim2user1", "scim2user2", "scim2user3", "scim2user4", "scim2user5"));
+    public static final List<String> PASSWORDS = new ArrayList<>(
+            Arrays.asList("testPassword1", "testPassword2", "testPassword3", "testPassword4", "testPassword5"));
+
+    private CloseableHttpClient client;
+
+    private static final String EQUAL = "+Eq+";
+    private static final String STARTWITH = "+Sw+";
+    private static final String ENDWITH = "+Ew+";
+    private static final String CONTAINS = "+Co+";
+    private static final String AND_OPERATION = "+and+";
+    private static final String GROUPS = "groups";
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        client = HttpClients.createDefault();
+    }
+
+    List<String> userIds = new ArrayList<>();
+    List<String> groupIds = new ArrayList<>();
+
+    private String adminUsername;
+    private String adminPassword;
+    private String tenant;
+
+    @Factory(dataProvider = "SCIM2UserConfigProvider")
+    public SCIM2MultiAttributeUserFilterTestCase(TestUserMode userMode) throws Exception {
+
+        AutomationContext context = new AutomationContext("IDENTITY", userMode);
+        this.adminUsername = context.getContextTenant().getTenantAdmin().getUserName();
+        this.adminPassword = context.getContextTenant().getTenantAdmin().getPassword();
+        this.tenant = context.getContextTenant().getDomain();
+    }
+
+    @DataProvider(name = "SCIM2UserConfigProvider")
+    public static Object[][] sCIM2UserConfigProvider() {
```

Comment:
Maintain the consistency when naming https://github.com/wso2/product-is/pull/3601/files#diff-cbb1ccc927c780601c2258b9b69028b5R109

Reply:
Fixed